proposals/lip-0058.md

```diff
@@ -13,58 +13,66 @@ Requires: 0055, 0056, 0061
 * `maxHeightCertified`: The maximum height of a block for which a certificate has been generated and included in the chain, see [LIP 0061][lip-0061]. This value is important for validating aggregate commits and for the [unlock transaction in DPoS][lip-0059].
 
 For computing the three values above, the following inputs are required:
 
-* **Prevote threshold**: This threshold value is required for correctly updating the property `maxHeightPrevoted`. This threshold can be computed from the sum of BFT weights of the active validators, see [LIP 0056][lip-0056] for details
+* **Prevote threshold**: This threshold value is required for correctly updating the property `maxHeightPrevoted`. This threshold can be computed from the sum of BFT weights of the active validators, see [LIP 0056][lip-0056] for details.
 * **Precommit threshold**: This threshold value is required for correctly updating the property `maxHeightPrecommitted`, see [LIP 0056][lip-0056] for details.
 * **Certificate threshold**: This threshold value is important for validating aggregate commits included in blocks and then updating the value of `maxHeightCertified`, see [LIP 0061][lip-0061].
 * **BFT weights of consensus participants**: For computing the prevote and precommit weight of a block, the BFT weight of the active validators at that block height is required.
 * **Block headers information**: A subset of the properties of the recent block headers in the current chain is required to update the prevote and precommits weights. The reason is that validators cast prevotes and precommits as part of their block header, see [LIP 0056][lip-0056] for details.
 
-We call the first four inputs above the **BFT parameters**. For the BFT module, we assume that these BFT parameters are provided from another module via the `setBFTParameters` function defined below. For a DPoS chain, the DPoS module would need to call the BFT module with the correct parameters, in particular, notifying the BFT module about any changes of validators. Similarly, for a PoA chain, it is the responsibility of the PoA module to provide the correct parameters as input to the BFT module. The block header information can be obtained by the module itself as part of the block processing.
+We call the first four inputs above the **BFT parameters**. We assume that these BFT parameters are forwarded from the state machine layer to the replication layer whenever the [setValidatorParams][lip-0044#setValidatorParams] function in the Validators module of the state machine layer is called. For a DPoS chain, the DPoS module would need to call this function with the correct parameters, in particular, notifying the replication layer about any changes of validators. Similarly, for a PoA chain, it is the responsibility of the PoA module to provide the correct parameters as input to `setValidatorParams` function. The block header information can be obtained during the block processing in the replication layer.
```

Comment:
I have some problems parsing the last sentence. I would find it clearer to just say smtg like: "The block header information is always available to the replication layer." or even be more specific about what is considered the 'engine-state'.